### PR TITLE
[MRG+1] Revise/modernize item exporter example in docs

### PR DESCRIPTION
**Pull request by colinmorris**
Addresses #2932

@kmike pointed out that the concept of a file-per-spider no longer made sense, so I changed the use case to one inspired by my own experience. It's maybe a little more complicated than the previous example, but it still comes out to fewer lines of code after removing the overhead of manually connecting the signals.

### CODE REVIEW

File: docs/topics/exporters.rst [hunk 1]
```diff
@@ -36,38 +36,36 @@ to export
...
+    from scrapy.exporters import XmlItemExporter
+
+    class PerYearXmlExportPipeline(object):
+        """Distribute items across multiple XML files according to their 'year' field"""
+
+        def open_spider(self, spider):
+            self.year_to_exporter = {}
+
+        def close_spider(self, spider):
+            for exporter in self.year_to_exporter.itervalues():
```
Comment (kmike) [inline docs/topics/exporters.rst]: It should be .values() in order to be compatible with Python 3. Creating a temporary list in Python 2 is not a problem as a list would be small.

File: docs/topics/exporters.rst [hunk 2]
```diff
@@ -36,38 +36,36 @@ to export
...
+                exporter.finish_exporting()
+                exporter.file.close()
+
+        def _exporter_for_item(self, item):
+            year = item['year']
+            if year not in self.year_to_exporter:
+                f = open('{}.xml'.format(year), 'w+b')
```
Comment (kmike) [inline docs/topics/exporters.rst]: why is 'w+b' needed, not just 'wb'?

**Comment by kmike** [comment]
I like the revised example, exporting to several locations is a common problem users ask for 👍 
The PR looks good to me, besides a couple of minor comments.

File: docs/topics/exporters.rst (hunk 2 above)
Comment (colinmorris) [reply docs/topics/exporters.rst]: Good call. I just copied it from the previous example, but as you say, it doesn't seem to be necessary.

**Comment by kmike** [comment]
Looks good to me, thanks @colinmorris!

**Pull request merged**